CondTools/Hcal/interface/instantiateEosTemplates.h

```diff
@@ -0,0 +1,74 @@
+#ifndef CondTools_Hcal_instantiateEosTemplates_h_
+#define CondTools_Hcal_instantiateEosTemplates_h_
+
+// Explicit instantiation of boost templates for eos archives.
+// Needed to overcome g++ quirks.
+#include <boost/archive/impl/basic_binary_oarchive.ipp>
+#include <boost/archive/impl/basic_binary_oprimitive.ipp>
+
+#if BOOST_VERSION < 104000
+#include <boost/archive/impl/archive_pointer_oserializer.ipp>
+#elif !defined BOOST_ARCHIVE_SERIALIZER_INCLUDED
+#include <boost/archive/impl/archive_serializer_map.ipp>
+#define BOOST_ARCHIVE_SERIALIZER_INCLUDED
+#endif
+
+namespace boost { namespace archive {
+
+	// explicitly instantiate for this type of binary stream
+	template class basic_binary_oarchive<eos::portable_oarchive>;
+
+	template class basic_binary_oprimitive<
+		eos::portable_oarchive
+	#if BOOST_VERSION < 103400
+		, std::ostream
+	#else
+		, std::ostream::char_type
+		, std::ostream::traits_type
+	#endif
+	>;
+
+#if BOOST_VERSION < 104000
+	template class detail::archive_pointer_oserializer<eos::portable_oarchive>;
+#else
+	template class detail::archive_serializer_map<eos::portable_oarchive>;
+	template class detail::archive_serializer_map<eos::polymorphic_portable_oarchive>;
+#endif
+
+} } // namespace boost::archive
+
+#include <boost/archive/impl/basic_binary_iarchive.ipp>
+#include <boost/archive/impl/basic_binary_iprimitive.ipp>
+
+#if BOOST_VERSION < 104000
+#include <boost/archive/impl/archive_pointer_iserializer.ipp>
+#elif !defined BOOST_ARCHIVE_SERIALIZER_INCLUDED
+#include <boost/archive/impl/archive_serializer_map.ipp>
+#define BOOST_ARCHIVE_SERIALIZER_INCLUDED
+#endif
+
+namespace boost { namespace archive {
```

Comment:
I don't think we should be adding things to the boost namespace -  is this code taken from somewhere?